specs/xaml-resource-references-tracing-spec.md

```diff
@@ -0,0 +1,191 @@
+strong indicator of an erroneous reference.
+
+Note: The below example output is for illustrative purposes only. The precise format of the message is 
+implementation-defined and may change in the future. Applications should not attempt to parse the message.
+
+```
+Beginning search for resource with key 'OutputTextBlockStyl'.
+  Searching dictionary 'ms-appx:///Controls/ControlExample.xaml' for resource with key 'OutputTextBlockStyl'.
+  Finished searching dictionary 'ms-appx:///Controls/ControlExample.xaml'.
+  Searching dictionary 'Framework-defined colors' for resource with key 'OutputTextBlockStyl'.
+  Finished searching dictionary 'Framework-defined colors'.
+  Searching dictionary 'Framework ThemeResources.xbf' for resource with key 'OutputTextBlockStyl'.
+    Searching theme dictionary (active theme: 'Light') for resource with key 'OutputTextBlockStyl'.
+      Searching dictionary '<anonymous dictionary>' for resource with key 'OutputTextBlockStyl'.
+      Finished searching dictionary '<anonymous dictionary>'.
+    Finished searching theme dictionary (active theme: 'Light').
+  Finished searching dictionary 'Framework ThemeResources.xbf'.
+  Searching dictionary 'ms-appx:///App.xaml' for resource with key 'OutputTextBlockStyl'.
+    Searching merged dictionary with index '1' for resource with key 'OutputTextBlockStyl'.
+      Searching dictionary 'ms-appx:///Microsoft.UI.Xaml/Themes/themeresources.xaml' for resource with key 'OutputTextBlockStyl'.
+        Searching theme dictionary (active theme: 'Light') for resource with key 'OutputTextBlockStyl'.
+          Searching dictionary 'ms-appx:///Microsoft.UI.Xaml/Themes/themeresources.xaml' for resource with key 'OutputTextBlockStyl'.
+          Finished searching dictionary 'ms-appx:///Microsoft.UI.Xaml/Themes/themeresources.xaml'.
+        Finished searching theme dictionary (active theme: 'Light').
+      Finished searching dictionary 'ms-appx:///Microsoft.UI.Xaml/Themes/themeresources.xaml'.
+    Finished searching merged dictionary with index '1'.
+    Searching merged dictionary with index '0' for resource with key 'OutputTextBlockStyl'.
+      Searching dictionary 'ms-appx:///ItemTemplates.xaml' for resource with key 'OutputTextBlockStyl'.
+      Finished searching dictionary 'ms-appx:///ItemTemplates.xaml'.
+    Finished searching merged dictionary with index '0'.
+    Searching theme dictionary (active theme: 'Light') for resource with key 'OutputTextBlockStyl'.
+      Searching dictionary 'ms-appx:///App.xaml' for resource with key 'OutputTextBlockStyl'.
+      Finished searching dictionary 'ms-appx:///App.xaml'.
+    Finished searching theme dictionary (active theme: 'Light').
+  Finished searching dictionary 'ms-appx:///App.xaml'.
+Finished search for resource with key 'OutputTextBlockStyl'.
+```
+
+
+## XamlResourceReferenceFailedEventArgs.Message Property
+
+A human-readable explanation of the XAML resource reference failure.
```

Comment:
Maybe "A human-readable explanation (in English) of the..."

Reply:
Is it the norm for our docs to indicate when a developer-facing string is not localized? In other words, is the fact that *this* message isn't localized notable enough to call out (and if it is, should we be localizing it after all)?

Reply:
I thought we localized even developer-facing strings. Because we have lots of developers who don't speak English natively.

Reply:
No, we don't localize debug strings. That's actually part of why the error message in RoOriginate doesn't make it out into the Exception.Message of the C# projection; so that unlocalized strings don't accidentally sneak out.

There's little/no precedent in the docs for this because we don't usually doc the debug strings. I think Raymond's tweak is a good idea.

Reply:
Thanks for the explanation! I've updated the line to match Raymond's suggestion.